src/pyconverter/xml2py/utils/utils.py

```diff
@@ -0,0 +1,162 @@
+    # convert all to flat and determine number of occurances
+    naive_names = []
+    rules = get_config_data_value(yaml_file_path, "rules")
+    specific_command_mapping = get_config_data_value(yaml_file_path, "specific_command_mapping")
+    for ans_name in meta_command:
+        ans_name = ans_name.lower()
+        if not ans_name[0].isalnum():
+            ans_name = ans_name[1:]
+        naive_names.append(ans_name)
+
+    # map command to pycommand function
+    name_map = {}
+
+    # second pass for each name
+    for ans_name in meta_command:
+        if ans_name in specific_command_mapping:
+            py_name = specific_command_mapping[ans_name]
+        else:
+            lower_name = ans_name.lower()
+            if not lower_name[0].isalnum():
+                alpha_name = lower_name[1:]
+            else:
+                alpha_name = lower_name
+
+            if naive_names.count(alpha_name) > 1:
+                if rules:
+                    py_name = lower_name
+                    for rule_name, rule in rules.items():
+                        py_name = py_name.replace(rule_name, rule)
+                    if py_name == lower_name and not py_name[0].isalnum():
+                        raise ValueError(
+                            f"""
+                            Additional rules need to be defined.
+                            The {ans_name} function name is in conflict with another function.
+                            """
+                        )
+                else:
+                    raise ValueError(
+                        f"Function '{ans_name}' has identical name to another function."
+                        "You need to provide RULES to differentiate them."
+                    )
+
+            else:
+                py_name = alpha_name
+
+        name_map[ans_name] = py_name
+
+    return name_map
+
+
+def import_handler(filename, additional_content, findalls):
+    needed_imports = ""
+    for match in findalls:
+        needed_imports += f"{match[0]}\n"
+        additional_content = additional_content.replace(match[0], "").replace("\n\n", "\n")
+
+    with open(filename, "r+") as f:
+        content = f.read()
+        f.seek(0, 0)
+        f.write(needed_imports + content + additional_content)
+
+
+# ############################################################################
+# AST functions
+# ############################################################################
+
+
+def split_trail_alpha(text):
+    """Split a string based on the last tailing non-alphanumeric character."""
+    for ii, char in enumerate(text):
+        if not char.isalnum():
+            break
+
+    ii += 1
+
+    return text[:ii], text[ii:]
+
+
+def is_numeric(text):
+    """Return ``True`` when a string is numeric."""
+    try:
+        float(text)
+        return True
+    except ValueError:
+        return False
+
+
+def get_refentry(filename):
+    """
+    Get the refentry from the XML file.
```

Comment:
```suggestion
    Get the reference entry from the XML file.
```
Descriptions should avoid abbreviations and use plain text.